### PR TITLE
devtoolbox: init at 1.2 (and dependencies)

### LINKED ISSUE #348015: Package request: Dev Toolbox
**Project description**

A collection of powerful yet simple-to-use tools and utilities to solve the most common daily development problems

**Metadata**

* homepage URL: https://flathub.org/apps/me.iepure.devtoolbox
* source URL: https://github.com/aleiepure/devtoolbox
* license: GPL-3.0+
* platforms: linux

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

### PR DESCRIPTION
**Pull request by ghost**
- Fixes #348015
- Base #295583
## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/development/python-modules/daltonlens/default.nix [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+  pythonOlder,
+  setuptools,
+  setuptools-git,
+  numpy,
+  pillow,
+}:
+buildPythonPackage rec {
+  pname = "daltonlens";
+  version = "0.1.5";
+  disabled = pythonOlder "3.7";
```
Comment (Aleksanaa) [inline pkgs/development/python-modules/daltonlens/default.nix:14]:
```suggestion
```

File: pkgs/development/python-modules/python-lorem/default.nix [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+  buildPythonPackage,
+  fetchPypi,
+  pythonOlder,
+  setuptools,
+  python3Packages,
+}:
+buildPythonPackage rec {
+  pname = "python-lorem";
+  version = "1.3.0.post3";
+  disabled = pythonOlder "3.5";
```
Comment (Aleksanaa) [inline pkgs/development/python-modules/python-lorem/default.nix:12]:
```suggestion
```

File: pkgs/development/python-modules/textstat/default.nix [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+  fetchFromGitHub,
+  pythonOlder,
+  setuptools,
+  pyphen,
+  python3Packages,
+}:
+buildPythonPackage rec {
+  version = "0.7.4";
+  pname = "textstat";
+  disabled = pythonOlder "3.6";
```
Comment (Aleksanaa) [inline pkgs/development/python-modules/textstat/default.nix:13]:
```suggestion
```

File: pkgs/development/python-modules/uuid6/default.nix [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+  buildPythonPackage,
+  fetchPypi,
+  pythonOlder,
+  setuptools,
+  python3Packages,
+}:
+buildPythonPackage rec {
+  pname = "uuid6";
+  version = "2024.7.10";
+  disabled = pythonOlder "3.8";
```
Comment (Aleksanaa) [inline pkgs/development/python-modules/uuid6/default.nix:12]:
```suggestion
```

**Comment by Aleksanaa** [comment]
The code is so beautiful 🙂↔️

File: pkgs/by-name/de/devtoolbox/package.nix [hunk 5]
```diff
@@ -0,0 +1,115 @@
...
+    homepage = "https://github.com/aleiepure/devtoolbox";
+    license = with lib.licenses; [
+      gpl3Plus
+      cc0
+      lgpl3Only
+      mit
+      unlicense
+    ];
+    mainProgram = "devtoolbox";
+    maintainers = with lib.maintainers; [ aleksana ];
```
Comment (Aleksanaa) [inline pkgs/by-name/de/devtoolbox/package.nix:112]: Add yourself to maintainers?

File: pkgs/development/python-modules/python-lorem/default.nix [hunk 6]
```diff
@@ -0,0 +1,34 @@
...
+  python3Packages,
+}:
+buildPythonPackage rec {
+  pname = "python-lorem";
+  version = "1.3.0.post3";
+  pyproject = true;
+
+  src = fetchPypi {
+    inherit version;
+    pname = "python_lorem";
```
Comment (Aleksanaa) [inline pkgs/development/python-modules/python-lorem/default.nix:15]:
```suggestion
```

Not sure why you would do this, https://pypi.org/project/python_lorem/ is redirected to https://pypi.org/project/python-lorem/

Comment (Aleksanaa) [reply pkgs/development/python-modules/python-lorem/default.nix:15]: oh okay, weird

**Review by Aleksanaa — APPROVED** [review]

File: pkgs/development/python-modules/uuid6/default.nix
Comment (Aleksanaa) [inline pkgs/development/python-modules/uuid6/default.nix:1]: This doesn't build:

```
=================================== FAILURES ===================================
_____________________________ UUIDTests.test_time ______________________________

self = <test_uuid6.UUIDTests testMethod=test_time>

    def test_time(self):
        uuid_1 = uuid1()
        uuid_6 = uuid6()
>       self.assertAlmostEqual(uuid_6.time / 10**7, uuid_1.time / 10**7, 3)
E       AssertionError: 13949846188.257402 != 13949846188.256878 within 3 places (0.0005245208740234375 difference)

test/test_uuid6.py:144: AssertionError
=========================== short test summary info ============================
FAILED test/test_uuid6.py::UUIDTests::test_time - AssertionError: 13949846188.257402 != 13949846188.256878 within 3 places (0...
========================= 1 failed, 27 passed in 0.27s =========================
```

File: pkgs/by-name/de/devtoolbox/package.nix [hunk 7]
```diff
@@ -0,0 +1,118 @@
...
+
+  buildInputs = [
+    libadwaita
+    gtksourceview5
+    webkitgtk_6_0
+    gcr_4
+    gdk-pixbuf
+  ];
+
+  propagatedBuildInputs = with python3Packages; [
```
Comment (Aleksanaa) [inline pkgs/by-name/de/devtoolbox/package.nix:48]:
```suggestion
  dependencies = with python3Packages; [
```

File: pkgs/development/python-modules/daltonlens/default.nix [hunk 8]
```diff
@@ -0,0 +1,40 @@
...
+    inherit pname version;
+    hash = "sha256-T7fXlRdFtcVw5WURPqZhCmulUi1ZnCfCXgcLtTHeNas=";
+  };
+
+  nativeBuildInputs = [
+    setuptools
+    setuptools-git
+  ];
+
+  propagatedBuildInputs = [
```
Comment (Aleksanaa) [inline pkgs/development/python-modules/daltonlens/default.nix:25]:
```suggestion
  dependencies = [
```

File: pkgs/development/python-modules/textstat/default.nix [hunk 9]
```diff
@@ -0,0 +1,45 @@
...
+    rev = version;
+    hash = "sha256-UOCWsIdoVGxmkro4kNBYNMYhA3kktngRDxKjo6o+GXY=";
+  };
+
+  nativeBuildInputs = [
+    setuptools
+    python3Packages.pytestCheckHook
+  ];
+
+  propagatedBuildInputs = [
```
Comment (Aleksanaa) [inline pkgs/development/python-modules/textstat/default.nix:26]:
```suggestion
  dependencies = [
```

File: pkgs/development/python-modules/daltonlens/default.nix [hunk 10]
```diff
@@ -0,0 +1,40 @@
...
+  pname = "daltonlens";
+  version = "0.1.5";
+  pyproject = true;
+
+  src = fetchPypi {
+    inherit pname version;
+    hash = "sha256-T7fXlRdFtcVw5WURPqZhCmulUi1ZnCfCXgcLtTHeNas=";
+  };
+
+  nativeBuildInputs = [
```
Comment (Aleksanaa) [inline pkgs/development/python-modules/daltonlens/default.nix:20]:
```suggestion
  build-system = [
```

File: pkgs/development/python-modules/python-lorem/default.nix [hunk 11]
```diff
@@ -0,0 +1,34 @@
...
+    hash = "sha256-Vw1TKheXg+AkhksnmWUfdIo+Jt7X7m1pS2f0Kfe8pv0=";
+  };
+
+  nativeBuildInputs = [
+    setuptools
+    python3Packages.pytestCheckHook
+  ];
```
Comment (Aleksanaa) [inline pkgs/development/python-modules/python-lorem/default.nix:22]:
```suggestion
  build-system = [
    setuptools
  ];

  nativeCheckInputs = [
    pytestCheckHook
  ];
```

function inputs of `python3Packages.callPackage` (which is the case for all in-tree `python3Packages` attributes) are the expansion of `python3Packages` on top of top-level packages.

File: pkgs/development/python-modules/python-lorem/default.nix [hunk 12]
```diff
@@ -0,0 +1,34 @@
+{
+  lib,
+  buildPythonPackage,
+  fetchPypi,
+  setuptools,
+  python3Packages,
```
Comment (Aleksanaa) [inline pkgs/development/python-modules/python-lorem/default.nix:6]:
```suggestion
  pytestCheckHook,
```

File: pkgs/development/python-modules/textstat/default.nix [hunk 13]
```diff
@@ -0,0 +1,45 @@
...
+    owner = "textstat";
+    repo = "textstat";
+    rev = version;
+    hash = "sha256-UOCWsIdoVGxmkro4kNBYNMYhA3kktngRDxKjo6o+GXY=";
+  };
+
+  nativeBuildInputs = [
+    setuptools
+    python3Packages.pytestCheckHook
+  ];
```
Comment (Aleksanaa) [inline pkgs/development/python-modules/textstat/default.nix:24]:
```suggestion
  build-system = [
    setuptools
  ];
```

File: pkgs/development/python-modules/textstat/default.nix [hunk 14]
```diff
@@ -0,0 +1,45 @@
...
+
+  propagatedBuildInputs = [
+    setuptools
+    pyphen
+  ];
+
```
Comment (Aleksanaa) [inline pkgs/development/python-modules/textstat/default.nix:30]:
```suggestion
  nativeCheckInputs = [
    pytestCheckHook
    pytest
  ];
```

File: pkgs/development/python-modules/textstat/default.nix [hunk 15]
```diff
@@ -0,0 +1,45 @@
+{
+  lib,
+  buildPythonPackage,
+  fetchFromGitHub,
+  setuptools,
+  pyphen,
+  python3Packages,
```
Comment (Aleksanaa) [inline pkgs/development/python-modules/textstat/default.nix:7]:
```suggestion
  pytestCheckHook,
  pytest,
```

File: pkgs/development/python-modules/uuid6/default.nix [hunk 16]
```diff
@@ -0,0 +1,33 @@
+{
+  lib,
+  buildPythonPackage,
+  fetchPypi,
+  setuptools,
+  python3Packages,
```
Comment (Aleksanaa) [inline pkgs/development/python-modules/uuid6/default.nix:6]:
```suggestion
  pytestCheckHook,
```

File: pkgs/development/python-modules/uuid6/default.nix [hunk 17]
```diff
@@ -0,0 +1,33 @@
...
+
+  src = fetchPypi {
+    inherit pname version;
+    hash = "sha256-LSnX9j9ZPKruoODQ3QrYEpycZjsp4ZvfiC6GS+3xj7A=";
+  };
+
+  nativeBuildInputs = [
+    setuptools
+    python3Packages.pytestCheckHook
+  ];
```
Comment (Aleksanaa) [inline pkgs/development/python-modules/uuid6/default.nix:21]:
```suggestion
  build-system = [
    setuptools
  ];

  nativeCheckInputs = [
    pytestCheckHook
  ];
```

**Comment by zendo** [comment]
I've tested and it works as expected.

**Pull request merged**